app/Chatkit Quickstart/ChatroomViewController.swift

```diff
@@ -1,48 +1,66 @@
 import UIKit
-// Import Chatkit dependency
+
+// Import the Chatkit SDK
 import PusherChatkit
 
+// For this example, you are adding all the Chatkit code to this View Controller.
+// In practice you might split the Chatkit logic across different files.
 class ChatroomViewController: UIViewController {
     
     @IBOutlet weak var messagesTableView: UITableView!
     @IBOutlet weak var textEntry: UITextField!
     
-    // Class params and inner classes
+    // Internal delegate class that listens to Chatkit connection-level events. Passed when initializing Chatkit.
+    // Implement it's methods to listen to different events
+    // https://pusher.com/docs/chatkit/reference/swift#pcchatmanagerdelegate
     class MyChatManagerDelegate: PCChatManagerDelegate {
         func onError(error: Error) {
             print("Error in Chat manager delegate! \(error.localizedDescription)")
         }
     }
     
+    // Chatkit properties
     public var chatManager: ChatManager?
     public var currentUser: PCCurrentUser?
     var messages = [PCMultipartMessage]()
     
     
+    // All initialization happens in this method
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        // Read the Chatkit instance details from Chatkit.plist
         guard let chatkitInfo = plistValues(bundle: Bundle.main) else { return }
         
         messagesTableView.delegate = self
         messagesTableView.dataSource = self
         
-        // Init Chatkit
+        // Instantiate Chatkit with instance ID, token provider endpoint, and ID of the user you're connecting as.
+        // Initialization: https://pusher.com/docs/chatkit/reference/swift#initialization
+        // Authenticating users and and providing tokens: https://pusher.com/docs/chatkit/reference/swift#pctokenprovider
         self.chatManager = ChatManager(
-            instanceLocator: chatkitInfo.instanceLocator,
-            tokenProvider: PCTokenProvider(url: chatkitInfo.tokenProviderEndpoint),
+            instanceLocator: chatkitInfo.instanceLocator, //Your Chatkit Instance ID
+            tokenProvider: PCTokenProvider(url: chatkitInfo.tokenProviderEndpoint), //Token provider endpoint
             userID: chatkitInfo.userId
         )
         
-        // Connect to Chatkit
+        // Connect to Chatkit by passing in the ChatManagerDelegate you defined at the
```

Comment:
at the… I think you meant to say something else here :)

Reply:
Yes, good spot! 🙈

Reply:
Just updated - I guess it makes more sense now :)